cql3/statements/describe_statement.hh

```diff
@@ -80,7 +80,7 @@ class schema_describe_statement : public describe_statement {
 
 private:
     std::variant<schema_desc, keyspace_desc> _config;
-    bool _with_internals;
+    [[maybe_unused]] bool _with_internals;
```

Comment:
But they are used. I tried to remove the field and found out that each class here uses it.
It seems like a clang bug.
`listing_describe_statement` is the only one that doesn't seem to use it. Maybe removing it from there makes the warning go away?
This code was merged very recently and we already moved to clang14. So we are not testing with clang12 anymore, you can expect more breakages with clang12, although there are developers who are still on clang12 and will probably keep patching scylla for clang12 support for some time.

Reply:
```diff
diff --git a/cql3/statements/describe_statement.cc b/cql3/statements/describe_statement.cc
index 4d2b631bbc..6513a56b27 100644
--- a/cql3/statements/describe_statement.cc
+++ b/cql3/statements/describe_statement.cc
@@ -579,10 +579,10 @@ future<std::vector<std::vector<bytes_opt>>> schema_describe_statement::describe(
 }
 
 // LISTING DESCRIBE STATEMENT
-listing_describe_statement::listing_describe_statement(element_type element, bool with_internals) 
+listing_describe_statement::listing_describe_statement(element_type element) 
     : describe_statement()
     , _element(element)
-    , _with_internals(with_internals) {}
+{}
 
 std::vector<lw_shared_ptr<column_specification>> listing_describe_statement::get_column_specifications() const {
     return get_listing_column_specifications();
@@ -701,7 +701,7 @@ std::unique_ptr<prepared_statement> describe_statement::prepare(data_dictionary:
             return ::make_shared<schema_describe_statement>(std::move(cfg.keyspace), cfg.only_keyspace, internals);
         },
         [&] (const describe_listing& cfg) -> ::shared_ptr<statements::describe_statement> {
-            return ::make_shared<listing_describe_statement>(cfg.element_type, internals);
+            return ::make_shared<listing_describe_statement>(cfg.element_type);
         },
         [&] (const describe_element& cfg) -> ::shared_ptr<statements::describe_statement> {
             return ::make_shared<element_describe_statement>(cfg.element_type, std::move(cfg.keyspace), std::move(cfg.name), internals);
@@ -783,4 +783,4 @@ std::unique_ptr<describe_statement> describe_statement::generic(std::optional<ss
 
 } // namespace statements
 
-} // namespace cql3
\ No newline at end of file
+} // namespace cql3
diff --git a/cql3/statements/describe_statement.hh b/cql3/statements/describe_statement.hh
index 80a4f2bf30..f20ae8885d 100644
--- a/cql3/statements/describe_statement.hh
+++ b/cql3/statements/describe_statement.hh
@@ -94,14 +94,13 @@ class schema_describe_statement : public describe_statement {
 class listing_describe_statement : public describe_statement {
 private:
     element_type _element;
-    bool _with_internals;
 
 protected:
     virtual std::vector<lw_shared_ptr<column_specification>> get_column_specifications() const override;
     virtual seastar::future<std::vector<std::vector<bytes_opt>>> describe(cql3::query_processor& qp, const service::client_state& client_state) const override;
 
 public:
-    listing_describe_statement(element_type element, bool with_internals);
+    listing_describe_statement(element_type element);
     listing_describe_statement(const listing_describe_statement&) = default;
 };
```
This patch compiles for me. Try applying it and see if the warning goes away.

Reply:
My impression was that there is "used" as in "passed around" (obviously) and there's "used" as in "the actual lowest-level function the value has been passed to actually looks at it" (no).  But the warning is still not exactly helpful even if my impression was correct, no argument there...

re: Clang versions, what I see with Clang as packaged in Nixpkgs is that both version 13 and 14 fail to compile Scylla in debug mode, same as described in #10909.  If the failure to build in debug mode is actually considered fine, I'll happily bump the Nix devenv to Clang 14.

Reply:
Indeed it doesn't seem to be used at all. @Jadw1 can we remove this variable altogether?

Reply:
Anyway, this does not need to hold up this series. I'll merge and @Jadw1 please have a look at this and send a patch if it can be removed. No need to keep dead code around.